Python/Tests/TestAdapterTests/TestExecutorTests.cs

```diff
@@ -188,6 +188,105 @@ class ManyTest(unittest.TestCase):
+        }
+
+
+        [TestMethod, Priority(0)]
+        [TestCategory("10s")]
+        public void RunPytestMissingPytestIdShowsErrorAndReturnsPartialResults() {
+            var testEnv = TestEnvironment.GetOrCreate(Version, FrameworkPytest);
+
+            var testFilePath1 = Path.Combine(testEnv.SourceFolderPath, "test_indirect_list.py");
+            File.Copy(TestData.GetPath("TestData", "TestDiscoverer", "Parameterized", "test_indirect_list.py"), testFilePath1);
+
+            var runSettings = new MockRunSettings(
+                new MockRunSettingsXmlBuilder(testEnv.TestFramework, testEnv.InterpreterPath, testEnv.ResultsFolderPath, testEnv.SourceFolderPath)
+                    .WithTestFilesFromFolder(testEnv.SourceFolderPath)
+                    .ToXml()
+            );
+            var discoveryContext = new MockDiscoveryContext(runSettings);
+            var discoverySink = new MockTestCaseDiscoverySink();
+            var logger = new MockMessageLogger();
+            var discoverer = new PythonTestDiscoverer();
+            discoverer.DiscoverTests(new[] { testFilePath1 }, discoveryContext, logger, discoverySink);
+
+            Console.WriteLine($"Discovered Tests");
+            foreach (var test in discoverySink.Tests) {
+                Console.WriteLine($"{test.DisplayName}");
+            }
+
+            Assert.IsTrue(discoverySink.Tests.Any());
+            Assert.AreEqual(discoverySink.Tests.Count(), 1);
+
+            // create a Missing Test with valid file but missing pytestId
+            var tc = discoverySink.Tests.First();
+            var validId = tc.GetPropertyValue<string>(Microsoft.PythonTools.TestAdapter.Pytest.Constants.PytestIdProperty, "");
+            var missingTest = new TestCase(tc.DisplayName + "_copy", tc.ExecutorUri, tc.Source);
+            missingTest.CodeFilePath = tc.CodeFilePath;
+            var missingPytestId = validId + "_copy";
+            missingTest.SetPropertyValue<string>(Microsoft.PythonTools.TestAdapter.Pytest.Constants.PytestIdProperty, missingPytestId);
+            discoverySink.Tests.Add(missingTest);
+
+            var testCases = discoverySink.Tests;
+            var runContext = new MockRunContext(runSettings, testCases, testEnv.ResultsFolderPath);
+            var recorder = new MockTestExecutionRecorder();
+            var executor = new TestExecutorPytest();
+            executor.RunTests(testCases, runContext, recorder);
+
+            //Check for Error Message
+            var errors = string.Join(Environment.NewLine, recorder.Messages);
+            AssertUtil.Contains(
+                errors,
+                "ERROR: not found:"
+            );
+
+            PrintTestResults(recorder);
+
+            //Check for Partial Results
+            Assert.IsTrue(recorder.Results.Any());
+            Assert.IsFalse(recorder.Results.Single( r => r.TestCase.DisplayName == discoverySink.Tests[0].DisplayName).Outcome != TestOutcome.Passed);
```

Comment:
some time later (not now, unless we need to push another commit for some reason) should be changed to Assert.AreEqual(TestOutcome.Passed, ...)

Reply:
ah good point